hathor/p2p/protocol.py

```diff
@@ -182,11 +182,40 @@ def change_state(self, state_enum: PeerState) -> None:
             if self.state:
                 self.state.on_enter()
 
+    def is_state(self, state_enum: PeerState) -> bool:
+        return isinstance(self.state, state_enum.value)
+
+    def get_short_remote(self) -> str:
```

Comment:
I initially liked this strategy, but after understanding that it needs `update_log_context` to be called every time any information used to generate this short description changes, I think it's weird.

I'd lean more to using separate properties even if it gets a little more verbose, our logs are structured enough that we can start forwarding them to a central logger, and being able to filter by `short_peer_id` without having to worry that the param is mixed with `host/port` or the connection state would be a big plus. In this case we would just keep binding these properties as soon as they're available:

```
self.log = self.log.bind(remote_id=self.get_short_peer_id())
# ...
self.log = self.log.bind(remote_host=f'{host}:{port}')
# ...
self.log = self.log.bind(connected=self.connected)  # either 1/0, or True/False
```

What do you think?

Reply:
I agree. I'll do that. Thanks!

Reply:
Fixed. Can you review again, please?

Reply:
Much better!